plugin/collectd-setup

```diff
@@ -0,0 +1,70 @@
+#!/bin/bash
+#
+# collectd-setup Startup script to setup collectd-mlab switch configuration.
+# chkconfig: - 85 16
+# description: collectd-setup performs auto-discovery of the local switch
+#   configuration to complete collectd setup. collectd-setup should run before
+#   collectd.
+
+
+# NOTE: Perform SNMP auto-discovery for switch configuration.
+# If we cannot update the snmp community string, then the script should halt.
+COLLECTD_SNMP="/etc/collectd-snmp.conf"
+COMMUNITY="/home/mlab_utility/config/snmp.community"
+SWITCH="s1.${HOSTNAME#*.}"
+
+function start() {
+
+    # If the community string file is missing or zero size.
+    if ! test -s "${COMMUNITY}" ; then
```

Comment:
Good question. The logic is a little particular.

The expected order of operations are:

* slice reinstalled
* community string distributed by ansible or restored by slice data backup.
* disco_config generates an updated community string.
* disco_config generates an snmp configuration for collectd.
* collectd collects switch traffic data.

At this point, if the community string has not been distributed yet, then the check `! test -s "${COMMUNITY}"` will be true and the error will be printed. Once it is present, we can proceed to the community string update.

What would help make this clearer?